orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+    ('mesh_headings', 'MH'),
+    ('abstract', 'AB'),
+    ('pub_date', 'DP'),
+]
+
+
+def _join_if_list(value):
+    if isinstance(value, list):
+        return ' '.join(value)
+    return value
+
+
+def _mesh_headings_to_class(mesh_headings):
+    """Extract a class from the mesh headings.
+
+    We take the first mesh heading and extract it to use as a class.
+
+    Args:
+        mesh_headings (str): The string containing all the mesh headings.
+
+    Returns:
+        str: The class value.
+    """
+    # e.g. heading1/heading2,heading3/*heading4
+    try:
+        regex = re.compile(r'^(.\w*?)\W')
+        class_mesh_groups = regex.search(mesh_headings[0])
+        return class_mesh_groups.groups()[0]
+    except:
+        return None
+
+
+def _date_to_iso(date):
+    possible_date_formats = [
+        '%Y %b %d',
+        '%Y %b',
+        '%Y',
+    ]
+
+    season_mapping = {
+        'fall': 'Sep',
+        'winter': 'Dec',
+        'spring': 'Mar',
+        'summer': 'Jun',
+    }
+
+    for date_format in possible_date_formats:
+        try:
+            return datetime.strptime(date, date_format).date().isoformat()
+        except Exception:
+            continue  # Try the next format.
+
+    date = date.lower()
+    # Seasons to their respective months.
+    for season, month in season_mapping.items():
+        date = date.replace(season, month)
+    try:  # Try month range (e.g. Jan-Mar).
+        date = date.split('-')[0]  # 2015 Sep-Dec --> 2015 Sep
+        return datetime.strptime(
+                date,
+                possible_date_formats[1]
+        ).date().isoformat()
+    except Exception:
+        warnings.warn(
+                'Could not parse "{}" into a date.'.format(
+                    date
+                ),
+                RuntimeWarning
+        )
+        return None
+
+
+def _records_to_corpus_entries(records, includes_metadata):
+    """Create corpus entries from records.
+
+    Args:
+        records (list): A list of dictionaries that hold record data.
+        includes_metadata (list): A list of tuples, where the
+            elements hold the names and tags of the metadata fields that we
+            wish to extract from the xml.
+
+    Returns:
+        list, list: Metadata and class values. Metadata is an array of size
+            n*m, where n is the number of article records contained within
+            'data' and m is the number of metadata fields we've chosen to
+            extract from the xml. The variable class_values is a list, where
+            the elements are class values for each article instance.
+    """
+    class_values = []
+    metadata = np.empty((len(records), len(includes_metadata)), dtype=object)
+
+    for row_num, record in enumerate(records):
+        fields = [
+                _join_if_list(record.get(field_key, ''))
+                for field_name, field_key
+                in includes_metadata
+            ]
+        fields[-1] = _date_to_iso(fields[-1])
```

Comment:
@nikicc If I remember correctly it's not up to the user to choose, we add it regardless and always at the last spot. But I do agree that negative indexing is not the best practice in this case and could lead to exceptions. Will look into this, we can probably handle it some other way.